antiplag/tasks.py

```diff
@@ -1,19 +1,61 @@
 from celery import shared_task
-from models import *
-from text_comparison import text_comparison
-import json
+
+from .models import Result
+
+from nlp.text_comparison import text_comparison
+from nlp.elastic import Elastic
+
+
+THRESHOLD = 0.15
+SIMILAR_COUNT = 10
+ERROR_MESSAGE = "An error occured during similarity comparison"
+
 
 @shared_task(name='antiplag.tasks.generate_result')
-def generate_result(document, elastic_result):
-    # One document to check against,
-    #up to 10 documents from elastic
-    corpus = document
-    corpus.append(elactic_result)
-    try:
-        result_json = text_comparison(document, corpus)
-    except:
-        antiplag_result = Result(document_id=document, error_msg="Comparison failed")
-        aantiplag_result.save()
-
-    antiplag_result = Result(document_id=document, matched_docs=result_json)
-    aantiplag_result.save()
+def compare_documents(documents):
+    """
+    Compare given documents against each other and N most similar documents from elastic
+    """
+
+    for doc in documents:
+        # returns list of dictionaries
+        # {
+        #   "document_name": "referaty-zemegula"
+        #   "text": "Co je zemegula? Je to hoax, zem je predsa plocha."
+        # }
+        similar_documents = Elastic.find_similar(doc.text, SIMILAR_COUNT)
+        # make new list and remove current
+        user_documents = list(documents).remove(doc)
+
+        results = []
+        # Compare current document with elastic docs
+        for similar_doc in similar_documents:
+            similarity = 0
+            try:
+                # returns percentage representing how similar docs are
+                similarity = text_comparison(doc.text, similar_doc["text"])
+            except:
+                # Ignore document
+                pass
+
+            if similarity > THRESHOLD:
+                results.append({
+                    "name": similar_doc["name"],
+                    "percentage": similarity
+                })
+
+        # Compare current document with user uploaded docs
+        for similar_doc in user_documents:
+            try:
+                # returns percentage representing how similar docs are
+                similarity = text_comparison(doc.text, similar_doc.text)
+            except:
+                similarity = {"error": ERROR_MESSAGE}
```

Comment:
Is there any point of appending uncomparable documents into matched docs?

If so, we should prepare frontend to support this as well.

This seems like a business oriented question and we should ask @juraj

Reply:
I'm not sure either. Should we keep it as is for now?

Reply:
Yeah, for now. Can you please add ad least # TODO: there ? So we can track it easily using IDE tools.